Add overview functionality

ryan1604 opened the pull request.

ryan1604 committed:
Add overview command

ryan1604 committed:
Update overview to show:

- Total balance
- Highest income
- Highest expense
- Watchlist

Fixes #29

ryan1604 commented:
dont merge yet

ryan1604 committed:
Merge branch 'master' into branch-overview

# Conflicts:
#	src/main/java/seedu/financialplanner/utils/Parser.java

ryan1604 committed:
Update overview command

ryan1604 committed:
Refactor code in budget and overview

Fixes #29

ryan1604 committed:
Fix bug in loading storage

ryan1604 committed:
Merge branch 'master' into branch-overview

# Conflicts:
#	src/main/java/seedu/financialplanner/utils/Parser.java

ryan1604 committed:
Update error handling for budget

NeoMinWei reviewed (dismissed):
deeply nested code from line 37 and 49 of overview command
maybe can implement rounding where user inputs value of 3d.p.?
other than that LGTM

ryan1604 committed:
Refactor code:

- Refactor formatting of double to string
- Reduce nesting in overviewcommand
- Clean up code

wwweert123 reviewed (approved):
LGTM, Maybe in the AddCashFlowCommand, you can create a method in UI to print the remaining budget or the exceeded budget instead of using showMessage

NeoMinWei reviewed (changes requested):
Very good in abstracting out the Ui elements
However, the code in lines 37 and 49 are STILL DEEPLY NESTED

ryan1604 commented:
> Very good in abstracting out the Ui elements However, the code in lines 37 and 49 are STILL DEEPLY NESTED

I believe 2-3 layers of nesting is fine

ryan1604 commented:
> LGTM, Maybe in the AddCashFlowCommand, you can create a method in UI to print the remaining budget or the exceeded budget instead of using showMessage

Good idea! I will look into it

ryan1604 committed:
Extract print statement to Ui

ryan1604 committed:
Merge branch 'master' into branch-overview

# Conflicts:
#	src/main/java/seedu/financialplanner/utils/Parser.java

NeoMinWei commented on src/main/java/seedu/financialplanner/list/Budget.java:
```diff
@@ -31,6 +33,16 @@ public static double getCurrentBudget() {
 
+    public static String getCurrentBudgetString() {
+        DecimalFormat decimalFormat = new DecimalFormat("####0.00");
+        return decimalFormat.format(Cashflow.round(currentBudget, 2));
```
good job on using rounding function to make the value more accurate when displayed in 2d.p.

NeoMinWei reviewed (commented):
LGTM
